image-service/src/storage.ts

```diff
@@ -0,0 +1,29 @@
+import crypto from "crypto";
+
+export interface IStorageAdapter {
+    save(oldPath: string, fileName: string): Promise<string>;
+}
+
+export interface IFile {
+    name: string;
+    type: string;
+    path: string;
+}
+
+export default class StorageService {
+
+    constructor(private adapter: IStorageAdapter) {
+    }
+
+    public async put(file: IFile): Promise<string> {
+        if (!file.type.startsWith("image/")) {
+            throw new Error(`Invalid file type! Expected: 'image/*', got '${file.type}'`);
+        }
+        return await this.adapter.save(file.path, this.encodeFilename(file.name));
+    }
+
+    public encodeFilename(fileName: string): string {
+        const [name, extension] = fileName.split(".");
```

Comment:
I'm using `name + Date.now()` so it should work fine for this purpose.

Reply:
For instance, if the user sends the image with some name `IMG_055.jpg` the result name will be `hash(IMG_055.jpg + Date.now())` = `hash(IMG_055.jpg + 1542637962296)`.
Then after 5 minutes, he sends the same file (it is the rare but possible situation). And we will have another name since `Date.now()` is different. In such a case, we will have two files with different names but with the same content.
But if we will save files with names as the content hash, we just will rewrite the existing file.

It's not critical. It's just a suggestion.

Reply:
let's move to the other ticket https://github.com/LvivNodeJsClub/parking-snitch/projects/2#card-15166033